File: selection_sort.py
#!/usr/bin/env python
print("enter numbers to sort")
nums = [int(x) for x in input().strip().split(" ")]

def find_min_and_swap(i):
    min = nums[i]
    minIndex = i
    for j in range(i+1, len(nums)):
        if nums[j] < min:
            min = nums[j]
            minIndex = j
    nums[minIndex] = nums[i]
    nums[i] = min

for i in range(len(nums)-1):
    find_min_and_swap(i)

print("sorted list: " + str(nums))

File: SelectionSort.java

public class SelectionSort {
	public int[] sort(int[] a) {
		int n = a.length;
		for (int i = 0; i < n - 1; i++) {
			int min = a[i];
			int min_index = i;
			for (int j = i + 1; j < n; j++) {
				if (a[j] < min) {
					min = a[j];
					min_index = j;
				}
			}
			a[min_index] = a[i];
			a[i] = min;
		}

		return a;
	}
}
